### PR TITLE
[#9] Add tripping function for canonical JSON instances

### LINKED ISSUE #9: Add round trip test functions for canonical JSON
This is required to bring over canonical JSON tests from the Byron release.

### PR DESCRIPTION
**Pull request by Jimbo4350**
This round trip function allows us to to verify that the `renderCanonicalJSON` and `parseCanonicalJSON` functions are coherent. This was inspired by the Byron release's canonical JSON tests found [here](https://github.com/input-output-hk/cardano-sl/blob/ccca8b346413f12e9fad8cab4c855e69669a2151/lib/src/Test/Pos/Helpers.hs)
Closes #9

### CODE REVIEW

File: stack.yaml [hunk 1]
```diff
@@ -3,3 +3,9 @@ resolver: ./snapshot.yaml
 packages:
   - .
   - test
+  # Library has never been released on Hackage yet
```
Comment (ruhatch) [inline stack.yaml]: We're adding stuff like this to the `snapshot.yaml` in this repo so they can be used in all repos. It should also be there already, so if you need something new just update the commit there.

File: test/Test/Cardano/Prelude/Golden.hs [hunk 2]
```diff
@@ -42,10 +42,10 @@ eachOf testLimit things hasProperty =
     >>= hasProperty
 
 goldenTestJSON
-  :: (Eq a, FromJSON a, HasCallStack, Show a, ToJSON a)
-  => a
-  -> FilePath
-  -> Property
+    :: (Eq a, FromJSON a, HasCallStack, Show a, ToJSON a)
```
Comment (ruhatch) [inline test/Test/Cardano/Prelude/Golden.hs]: Should stay as 2 spaces ;)

File: test/Test/Cardano/Prelude/Tripping.hs [hunk 3]
```diff
@@ -36,11 +71,62 @@ roundTripsAesonShow
...
+  => a
+  -> m ()
+roundTripsCanonicalJsonPretty a = tripping a canonicalEncPre canonicalDecPre
+ where
+  canonicalEncPre
+    :: forall a . CanonicalJSON.ToJSON Identity a => a -> LB.ByteString
+  canonicalEncPre x =
+      LB.fromStrict . encodeUtf8 . toS $
+      CanonicalJSON.prettyCanonicalJSON $ runIdentity $ CanonicalJSON.toJSON x
+  canonicalDecPre
```
Comment (ruhatch) [inline test/Test/Cardano/Prelude/Tripping.hs]: Don't need this definition twice if the pretty version doesn't do anything different

File: test/Test/Cardano/Prelude/Tripping.hs [hunk 4]
```diff
@@ -66,28 +152,27 @@ trippingBuildable
...
-        , "━━━ Roundtrip ━━━"
-        , show $ buildValue my
-        ]
-
-      Just dif ->
-        withFrozenCallStack
-          $ failWith
-              (Just $ Diff "━━━ " "- Original" "/" "+ Roundtrip" " ━━━" dif)
-          $ unlines ["━━━ Intermediate ━━━", show i]
+  let mx :: f a
```
Comment (ruhatch) [inline test/Test/Cardano/Prelude/Tripping.hs]: `brittany`s `IndentPolicyMultiple` will leave most of this function alone ;)

File: stack.yaml (hunk 1 above)
Comment (Jimbo4350) [reply stack.yaml]: Gotcha

**Pull request merged**